### PR TITLE
Refactored Grpc Gateway of Staking, TopUp, Clerk, Gov Module  

### PR DESCRIPTION
**Pull request by saikumar0x1967**
#558 
#559 
#560

Grpc Gateways 
1. Staking 
2. Checkpoint
3. Clerk
4. Gov

### CODE REVIEW

File: proto/heimdall/checkpoint/v1beta1/query.proto [hunk 1]
```diff
@@ -121,3 +125,11 @@ message QueryBorChainID {
...
     string bor_chainID = 1;
 }
+
+// QueryLatestCheckpointRequest is request for latest checkpoint
+message QueryLatestCheckpointRequest {}
+
+// Latest Checkpoint response
+message QueryLatestCheckpointResponse {
+    heimdall.types.Checkpoint latest_checkpoint = 1;
+}
```
Comment (murthyvitwit) [inline proto/heimdall/checkpoint/v1beta1/query.proto:135]: Need a new line at end of the file.

File: proto/heimdall/clerk/v1beta1/query.proto [hunk 2]
```diff
@@ -29,3 +39,22 @@ message QueryRecordParams {
...
+}
+
+// QueryRecordListRequest
+message QueryRecordListRequest {
+    heimdall.types.QueryPaginationParams pagination = 1;
+}
+
+message QueryRecordListResponse {
+    repeated EventRecord event_records = 1 [(gogoproto.nullable) = false];
+}
```
Comment (murthyvitwit) [inline proto/heimdall/clerk/v1beta1/query.proto:60]: Ditto

File: proto/heimdall/staking/v1beta1/query.proto [hunk 3]
```diff
@@ -48,3 +55,14 @@ message QueryValidatorSetResponse {
...
+// QueryStakingSequenceParams defines the params for querying an account
+// Sequence.
+message QueryStakingOldTxRequest {
+    string tx_hash   = 1 [(gogoproto.moretags) = "yaml:\"tx_hash\""];
+    uint64 log_index = 2 [(gogoproto.moretags) = "yaml:\"log_index\""];
+}
+
+message QueryStakingOldTxResponse {
+    bool status = 1;
+}
```
Comment (murthyvitwit) [inline proto/heimdall/staking/v1beta1/query.proto:68]: Ditto

File: proto/heimdall/staking/v1beta1/query.proto [hunk 4]
```diff
@@ -16,20 +16,27 @@ service Query {
...
+        returns (QueryStakingOldTxResponse) {
+        option (google.api.http).get = "/heimdall/staking/v1beta1/isoldtx";
     }
 }
 
 // QueryValidatorRequest is request type for the Query/Validator RPC method
 message QueryValidatorRequest {
     // validator_id defines the validator id to query for.
-    heimdall.types.ValidatorID validator_id = 1;
+    int32 validator_id = 1;
```
Comment (murthyvitwit) [inline proto/heimdall/staking/v1beta1/query.proto:39]: Why can't we use ValidatorID? @saikumar0x1967

File: x/clerk/module.go [hunk 5]
```diff
@@ -48,10 +47,6 @@ func (AppModuleBasic) Name() string {
 	return types.ModuleName
 }
 
-func (AppModuleBasic) RegisterCodec(cdc *codec.LegacyAmino) {
```
Comment (murthyvitwit) [inline x/clerk/module.go:51]: why did you remove this?

File: x/gov/module.go [hunk 6]
```diff
@@ -56,10 +53,6 @@ func (AppModuleBasic) Name() string {
 	return types.ModuleName
 }
 
-func (AppModuleBasic) RegisterCodec(cdc *codec.LegacyAmino) {
```
Comment (murthyvitwit) [inline x/gov/module.go:59]: Ditto

File: x/staking/client/cli/query.go [hunk 7]
```diff
@@ -45,7 +44,7 @@ func GetValidatorInfoCmd() *cobra.Command {
 				return err
 			}
 
-			validatorID := hmTypes.ValidatorID(viper.GetInt64(FlagValidatorID))
+			validatorID := viper.GetInt32(FlagValidatorID)
```
Comment (murthyvitwit) [inline x/staking/client/cli/query.go:47]: Do you have any strong reasons to avoid the ValidatorID type?

**Review by murthyvitwit — COMMENTED** [review]
Mostly looks good, Please check the comments and fix them.

File: x/staking/client/cli/query.go (hunk 7 above)
Comment (saikumar0x1967) [reply x/staking/client/cli/query.go:47]: Yes when we query validatorId in routes `validator/{validatorID}` ,  it is throwing invalid type for some reason so i used `int32` in proto

File: x/clerk/module.go (hunk 5 above)
Comment (saikumar0x1967) [reply x/clerk/module.go:51]: in AppModuleBasic interface there is no  `RegisterCodec` func

File: proto/heimdall/staking/v1beta1/query.proto (hunk 4 above)
Comment (saikumar0x1967) [reply proto/heimdall/staking/v1beta1/query.proto:39]: validator id not working in query params of grpc gateway routes

**Comment by saikumar0x1967** [comment]
@murthyvitwit  review changes are done. Please review it again

**Review by murthyvitwit — APPROVED** [review]
LGTM
